mobile/application-manager-base.ts

```diff
@@ -27,29 +28,46 @@ export abstract class ApplicationManagerBase extends EventEmitter implements Mob
 		}).future<boolean>()();
 	}
 
+	private isChecking = false;
 	public checkForApplicationUpdates(): IFuture<void> {
 		return (() => {
-			let newInstalledApplications = this.getInstalledApplications().wait();
-			let previouslyInstalledApplications = this.lastInstalledApplications || [];
+			// As this method is called on 500ms, but it's execution may last much longer
+			// use locking, so the next executions will not get into the body, while the first one is still working.
+			// In case we do not break the next executions, we'll report each app as newly installed several times.
+			if(!this.isChecking) {
+				this.isChecking = true;
+				let currentlyInstalledApplications = this.getInstalledApplications().wait();
```

Comment:
this variable actually contains the application's identifiers as opposed to the whole applications altogether. Could we change `currentlyInstalledApplications` and `previouslyInstalledApplications` to something like `currentlyInstalledApplicationIdentifiers` and `previouslyInstalledApplicationIdentifiers` respectively